Examples/CliClient/Program.cs

```diff
@@ -289,13 +293,46 @@ async Task PerformRequest(Connection conn)
     }
 
     /// <summary>Create a HTTP/2 connection to the remote peer</summary>
-    Task<Connection> CreateConnection(string host, int port)
+    Task<Connection> CreateConnection(string scheme, string host, int port)
     {
-        if (useHttp1Upgrade) return CreateUpgradeConnection(host, port);
-        else return CreateDirectConnection(host, port);
+        if (useHttp1Upgrade) return CreateUpgradeConnection(scheme, host, port);
+        else return CreateDirectConnection(scheme, host, port);
     }
 
-    async Task<Connection> CreateDirectConnection(string host, int port)
+    async Task<(IReadableByteStream, IWriteAndCloseableByteStream)> CreateStreams(string scheme, string host, int port)
+    {
+        // Create a TCP connection
+        logger.LogInformation($"Starting to connect to {scheme}://{host}:{port}");
+        var tcpClient = new TcpClient();
+        await tcpClient.ConnectAsync(host, port);
+        logger.LogInformation("Connected to remote");
+        tcpClient.Client.NoDelay = true;
+        if (scheme == "https")
+        {
+            var stream = new SslStream(tcpClient.GetStream());
+            logger.LogInformation("Negotiating SSL...");
+            var options = new SslClientAuthenticationOptions
+            {
+                TargetHost = host,
+                ApplicationProtocols = new List<SslApplicationProtocol>
+                {
+                    SslApplicationProtocol.Http2
+                },
+            };
+            await stream.AuthenticateAsClientAsync(options, default(CancellationToken));
+            logger.LogInformation("SSL Authenticated");
```

Comment:
Does at this place a verification need to get inserted that checks whether the server accepted the proposed ALPN protocol?

Reply:
I added a protocol verification.